assets/src/components/with-initial-positioning.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * WordPress dependencies
+ */
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { createBlock } from '@wordpress/blocks';
+
+/**
+ * External dependencies
+ */
+import { every } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { ALLOWED_CHILD_BLOCKS } from '../constants';
+import { withParentBlock } from './';
+
+const positionTopLimit = 75;
+
+export default createHigherOrderComponent(
+	( BlockEdit ) => {
+		return withParentBlock( ( props ) => {
+			const { attributes, name, parentBlock } = props;
+
+			if ( -1 === ALLOWED_CHILD_BLOCKS.indexOf( name ) || ! parentBlock || 'amp/amp-story-page' !== parentBlock.name ) {
+				return <BlockEdit { ...props } />;
+			}
+
+			// If the positions are not set and if the block is selected.
```

Comment:
@swissspidy -- this is the logic for initial positioning of a newly added block. Let me know if it seems to make sense to you, I was investigating the options of doing it in a different filter and a better way to detect if a block is unmodified but there doesn't seem to be one. Thought at first of `blocks.getBlockAttributes` filter but that doesn't fire for new blocks.

Reply:
Hmm we already have some logic in `amp-story-editor-blocks.js` to detect if there are any newly inserted pages. We can just extend this (e.g. using `getClientIdsWithDescendants()`) to check if there are _any_ newly inserted _blocks_.

If so, `updateBlockAttributes()` can be used to update the attributes of a newly inserted block. IMO that makes more sense than filtering `BlockEdit` and modifying attributes in it.